.github/workflows/ci.yml

```diff
@@ -0,0 +1,36 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches: main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        stack-yaml:
+          - stack-nightly.yaml   # ghc-9.2
+          - stack.yaml           # ghc-9.0
+          - stack-lts-18.28.yaml # ghc-8.10
```

Comment:
We can't go lower without a lot of extra-deps. `monad-logger-aeson` has poor compatibility, so we inherit that.